### PR TITLE
fix(Search): clicking bx--search-close icon invokes onChange

### LINKED ISSUE #101: Allow click on bx--search-icon to invoke onChange
I need to attach my own click handler logic to the bx--search--close icon that's used on `Search` component. Although it does clear input value, I also want it to handle some `this.setState` logic as well.

### PR DESCRIPTION
**Pull request by hellobrian**
resolves #101 

Mimics native `<input type="search">` close icon. 
Now clicking close icon will invoke `change` event.

### CODE REVIEW

**Comment by chrisdhanaraj** [comment]
Out of curiosity, why use this vs the normal `evt.target.value`? If evt.target.value === ' ' onChange, execute? 

(as I keep typing this, this probably is more clear though)

**Comment by hellobrian** [comment]
Hold up lol, as soon as I submitted this, I realized it might be better to have `clearInput` invoke `this.props.onChange(evt)`.

File: components/Search.js [hunk 1]
```diff
@@ -45,6 +46,7 @@ class Search extends Component {
       },
       () => this.input.focus()
     );
+    this.props.onChange(evt);
```
Comment (chrisdhanaraj) [inline components/Search.js]: So slotting this in here is actually running it twice for people who actually use the onChange input (once on the section where we wipe out the value, again here). I think what should happen is that you should merge the two objects - `evt` and then the faux `evt` object I have up there and then run the this.props.onchange once. You'd have to reverse the logic up there as well. It'd look something like

```js
if (!this.props.value) {
   this.input.value
} 

...

const newEvt = Object.assign(evt, { target: { value: '' } });
this.props.onChange(newEvt)
```

**Comment by chrisdhanaraj** [comment]
👍 ✅

**Comment by tw15egan** [comment]
My computer hangs each time I click the close button?

**Comment by marijohannessen** [comment]
Same happens for me ^^

File: components/Search.js [hunk 2]
```diff
@@ -14,54 +14,52 @@ class Search extends Component {
...
+    hasContent: this.props.value || this.props.defaultValue || false
   };
 
-  clearInput = () => {
-    if (this.props.value) {
-      this.props.onChange({
+  clearInput = evt => {
+    if (!this.props.value) {
+      this.input.value = '';
+      this.props.onChange(evt);
```
Comment (hellobrian) [inline components/Search.js]: @tw15egan and @marijohannessen pointed out that using `clearInput` was not firing `onChange` event in uncontrolled Search. So in this conditional that checks for controlled or uncontrolled, I added `this.props.onChange(evt)`. I'm not seeing duplicate `onChange` events being fired so I think this is ok. 

@chrisdhanaraj can you take on more look?

**Comment by chrisdhanaraj** [comment]
Bluemix [toolchain](https://console-regional.ng.bluemix.net/devops/toolchains/6b8b7163-1ec8-4616-82fa-aa467a17af03 "storybook-toolchain-v2"): [Delivery Pipeline](https://console-regional.ng.bluemix.net/devops/pipelines/b84aaac8-8484-4a1e-902a-ebb526eb7998 "storybook-toolchain-v2") deployed [carbon-components-react](https://github.com/carbon-design-system/carbon-components-react/releases/tag/deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20170802-190905 "deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20170802-190905") to [Bluemix Design System](https://console.bluemix.net/apps/0a253845-97bd-46c9-aa04-a220aeb8ed06?env_id=ibm:yp:us-south "ibm:yp:us-south:chrisdhanaraj_org:Bluemix Design System"), including commit bec57cb0e46c97a87b77d0389d4b40b1260365a3

**Pull request merged**